components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -3076,28 +3093,54 @@ def buildQuery(self, obj_type, ids=None, params=None, attributes=None):
                 "getObjects uses a string to define obj_type, E.g. "
                 "'Image' not %r" % obj_type)
 
-        if params is None:
-            params = omero.sys.Parameters()
-        if params.map is None:
-            params.map = {}
+        owner = None
+        order_by = None
+        opts = None
+        inputParams = None
+
+        if isinstance(params, dict):
```

Comment:
Adding `opts` as a keyword argument is certainly preferable from my perspective. @aleksandra-tarkowska would probably like the opportunity to comment.

I think you can choose one and document it as taking precedence. Whichever one is the subset of the other is probably the a good candidate but I can see arguments both ways. That is, if you have the possibility to configure 50 things with `opts` and have `params` only able to configure 10 the settings from `params` should probably override those set in `params` if a call were made where both were present.